examples/tree_source/tree_source/app.py

```diff
@@ -1,119 +1,150 @@
+            def sort_func(child):
+                return child._mtime
+        elif accessor == "name":
+            def sort_func(child):
+                return child.name[1].lower()
+        else:  # use the function provided by the user / default
+            def sort_func(child):
+                # sort according to value of accessor, using the provided sort key
+                try:
+                    attr = getattr(child, accessor)
+                    if isinstance(attr, tuple):
+                        icon, value = attr
+                    else:
+                        value = attr
+                    return key(value) if key else value
+                except AttributeError:
+                    return ''
+
+        # sort all children in hirarchy
+        self._children.sort(key=sort_func, reverse=reverse)
+
+        for c in self._children:
+            try:
+                c.sort(accessor, key, reverse)
+            except AttributeError:
+                pass
+
+
+class FileSystemSource(Node, Source):
+
+    def __init__(self, path):
+        super().__init__(path, parent=self)
+        self.path = path
+        self._parent = None
+        self._children = []
 
 
 class ExampleTreeSourceApp(toga.App):
-    # Table callback functions
-    def on_select_handler(self, widget, node):
-        if node and hasattr(node, 'title'):
-            self.label.text = 'You selected node: {}'.format(node.title)
-        else:
-            self.label.text = 'No row selected'
-
-    # Button callback functions
-    def insert_handler(self, widget, **kwargs):
-        entry = choice(bee_movies)
-        self.tree.data.add(entry)
 
     def startup(self):
         # Set up main window
         self.main_window = toga.MainWindow(title=self.name)
 
-        # Label to show responses.
-        self.label = toga.Label('Ready.')
+        root = os.path.abspath(os.sep)
+        self.fs_source = FileSystemSource(root)
 
         self.tree = toga.Tree(
-            headings=['Year', 'Title', 'Rating', 'Genre'],
-            data=DecadeSource(),
-            on_select=self.on_select_handler,
-            style=Pack(flex=1)
+            headings=['Name', 'Date Modified'],
+            data=self.fs_source,
+            style=Pack(flex=1),
+            multiple_select=True,
+            sorting=True,
+            sort_keys=None,
         )
 
-        # Buttons
-        btn_style = Pack(flex=1)
-        btn_insert = toga.Button('Insert Row', on_press=self.insert_handler, style=btn_style)
-        btn_box = toga.Box(children=[btn_insert], style=Pack(direction=ROW))
-
         # Outermost box
         outer_box = toga.Box(
-            children=[btn_box, self.tree, self.label],
-            style=Pack(
-                flex=1,
-                direction=COLUMN,
-                padding=10,
-            )
+            children=[
+                toga.Label('A list of all your folders!', style=Pack(padding=10)),
```

Comment:
It's implied here, but to avoid any possible confusion, we may want to add a comment that files *won't* be displayed. Even a joke "want your files? Use a file manager!" comment would do.

Reply:
I am German so I don't do jokes. But I have added a very serious sounding comment at the beginning, explaining what this example does.

Reply:
Fair enough :-)